test/metabase/models/dashboard_test.clj

```diff
@@ -29,168 +30,204 @@
-          (revision/diff-strings
-            Dashboard
-            {:cards [{:id 1 :row 0} {:id 2 :row 1}]}
-            {:cards [{:id 1 :row 1} {:id 2 :row 2}]}))))
-
- (is (= "modified the cards."
-        (build-sentence
-          (revision/diff-strings
-            Dashboard
-            {:cards [{:id 1} {:id 2}]}
-            {:cards [{:id 1} {:id 3}]}))))
-
- (is (= "renamed it from \"Apple\" to \"Next\" and modified the cards."
-        (build-sentence
-          (revision/diff-strings
-            Dashboard
-            {:name "Apple"
-             :cards [{:id 1} {:id 2}]}
-            {:name "Next"
-             :cards [{:id 1} {:id 3}]}))))
- (t2.with-temp/with-temp
-   [Collection {coll-id :id} {:name "New collection"}]
-   (is (= "moved this Dashboard to New collection."
+             {:cards [{:id 1} {:id 2}]}
+             {:cards [{:id 1} {:id 3}]})))))
+
+ (testing "update collection ---"
+  (t2.with-temp/with-temp
+    [Collection {coll-id :id} {:name "New collection"}]
+    (is (= "moved this Dashboard to New collection."
+           (build-sentence
+             (revision/diff-strings
+               Dashboard
+               {:name "Apple"}
+               {:name          "Apple"
+                :collection_id coll-id})))))
+  (t2.with-temp/with-temp
+    [Collection {coll-id-1 :id} {:name "Old collection"}
+     Collection {coll-id-2 :id} {:name "New collection"}]
+    (is (= "moved this Dashboard from Old collection to New collection."
+           (build-sentence
+             (revision/diff-strings
+               Dashboard
+               {:name          "Apple"
+                :collection_id coll-id-1}
+               {:name          "Apple"
+                :collection_id coll-id-2}))))))
+
+ (testing "update tabs"
+   (is (= "added a tab."
           (build-sentence
             (revision/diff-strings
               Dashboard
-              {:name "Apple"}
-              {:name          "Apple"
-               :collection_id coll-id})))))
- (t2.with-temp/with-temp
-   [Collection {coll-id-1 :id} {:name "Old collection"}
-    Collection {coll-id-2 :id} {:name "New collection"}]
-   (is (= "moved this Dashboard from Old collection to New collection."
+              {:tabs [{:id 0 :name "First tab" :position 0}]}
+              {:tabs [{:id 0 :name "First tab" :position 0}
+                      {:id 1 :name "Second tab" :position 1}]}))))
+
+   (is (= "removed 2 tabs."
           (build-sentence
             (revision/diff-strings
               Dashboard
-              {:name          "Apple"
-               :collection_id coll-id-1}
-              {:name          "Apple"
-               :collection_id coll-id-2}))))))
+              {:tabs [{:id 0 :name "First tab" :position 0}
+                      {:id 1 :name "Second tab" :position 1}
+                      {:id 2 :name "Third tab" :position 2}]}
+              {:tabs [{:id 0 :name "First tab" :position 0}]}))))
 
+   (is (= "rearranged the tabs."
```

Comment:
You need to add test cases for "modified the tabs" - both for renaming the tabs and adding and removing them at the same time

Reply:
added 0cbbda7595